tests/sentry/api/endpoints/test_project_rules.py

```diff
@@ -2,35 +2,54 @@
 from unittest.mock import patch
 
 import responses
-from django.urls import reverse
 
 from sentry.models import Environment, Integration, Rule, RuleActivity, RuleActivityType
 from sentry.testutils import APITestCase
 from sentry.utils import json
 
 
-class ProjectRuleListTest(APITestCase):
-    def test_simple(self):
-        self.login_as(user=self.user)
-
-        team = self.create_team()
-        project1 = self.create_project(teams=[team], name="foo")
-        self.create_project(teams=[team], name="bar")
+class ProjectRuleBaseTestCase(APITestCase):
+    endpoint = "sentry-api-0-project-rules"
 
-        url = reverse(
-            "sentry-api-0-project-rules",
-            kwargs={"organization_slug": project1.organization.slug, "project_slug": project1.slug},
+    def setUp(self):
+        self.user = self.create_user()
+        self.organization = self.create_organization(owner=self.user)
+        self.project = self.create_project(organization=self.organization)
```

Comment:
These three lines are already run in fixtures so you can omit them to save the DB roundtrips.
```py
self.user = self.create_user()
self.organization = self.create_organization(owner=self.user)
self.project = self.create_project(organization=self.organization)
```

Reply:
I will remove it since it's an established pattern but TBH I don't love having logic that tests not being in the same file :(